src/tools/compiletest/src/runtest.rs

```diff
@@ -2325,15 +2331,72 @@ impl<'test> TestCx<'test> {
             let res = self.cmd2procres(
                 Command::new(&self.config.docck_python)
                     .arg(root.join("src/etc/htmldocck.py"))
-                    .arg(out_dir)
+                    .arg(&out_dir)
                     .arg(&self.testpaths.file),
             );
             if !res.status.success() {
-                self.fatal_proc_rec("htmldocck failed!", &res);
+                self.fatal_proc_rec_with_ctx("htmldocck failed!", &res, |mut this| {
+                    this.compare_to_default_rustdoc(&out_dir)
+                });
             }
         }
     }
 
+    fn compare_to_default_rustdoc(&mut self, out_dir: &Path) {
+        println!("info: generating a diff against nightly rustdoc");
+
+        let suffix =
+            self.safe_revision().map_or("nightly".into(), |path| path.to_owned() + "-nightly");
+        let compare_dir = output_base_dir(self.config, self.testpaths, Some(&suffix));
+        let _ = fs::remove_dir_all(&compare_dir);
```

Comment:
What if this failed to remove directory even if it existed?

Reply:
A `expect` would be appreciated I guess. :)

Reply:
This is to match the previous behavior in `run_rustdoc_test`: https://github.com/rust-lang/rust/blob/5b1bd121f3e491548fe9c92a615a588328b9c3e0/src/tools/compiletest/src/runtest.rs#L685

I think this will give an error if `compare_dir` didn't already exist, so I don't think panicking on error is the right thing to do.

Reply:
Should we keep a comment to explain why it doesn't handle error?